two_factor/migrations/0003_auto_20150817_1733.py

```diff
@@ -0,0 +1,41 @@
+# -*- coding: utf-8 -*-
+from __future__ import unicode_literals
+import logging
+
+from django.db import models, migrations
+import phonenumbers
+import two_factor.models
+
+logger = logging.getLogger(__name__)
+
+
+def migrate_phone_numbers(apps, schema_editor):
+    PhoneDevice = apps.get_model("two_factor", "PhoneDevice")
+    for device in PhoneDevice.objects.all():
+        try:
+            number = phonenumbers.parse(device.number)
+            if not phonenumbers.is_valid_number(number):
+                logger.info("User '%s' has an invalid phone number '%s'." % (device.user.username, device.number))
+            device.number = phonenumbers.format_number(number, phonenumbers.PhoneNumberFormat.E164)
```

Comment:
Theoretically this shouldn't have happened, as Twilio also validates phone numbers itself. However I wanted to cover the edge-case of logging any invalid phone numbers instead of throwing an exception. I'm not sure how this can be handled more gracefully.